hopli/src/safe_module.rs

```diff
@@ -608,6 +662,81 @@ impl SafeModuleSubcommands {
         }
         Ok(())
     }
+
+    /// Execute the command to debug the following:
+    /// 1. node xDAI balance
+    /// 2. If node has been included on Network Registry
+    /// 3. If node and safe are associated on Node Safe Registry
+    /// 4. If Safe is owned by the correct owner(s)
+    /// 5. Safe’s wxHOPR balance and allowance
+    /// 6. if node is included in the module
+    /// 7. If the channel contract is included as a target
+    /// 8. If the announce contract is included as a target
+    /// 9. If safe is the owner of the module
+    #[allow(clippy::too_many_arguments)]
+    pub async fn execute_safe_module_debugging(
+        network_provider: NetworkProviderArgs,
+        local_identity: IdentityFileArgs,
+        node_address: Option<String>,
+        safe_address: String,
+        module_address: String,
+    ) -> Result<(), HelperErrors> {
+        // read all the node addresses
+        let mut node_eth_addresses: Vec<H160> = Vec::new();
+        if let Some(addresses) = node_address {
+            node_eth_addresses.extend(addresses.split(',').map(|addr| H160::from_str(addr).unwrap()));
+        }
+        // if local identity dirs/path is provided, read addresses from identity files
+        node_eth_addresses.extend(local_identity.to_addresses().unwrap().into_iter().map(H160::from));
+
+        // parse safe and module addresses
+        let safe_addr = H160::from_str(&safe_address).unwrap();
+        let module_addr = H160::from_str(&module_address).unwrap();
```

Comment:
Pls use proper error forwarding via `.map_err(...)?` instead of unwrap.

Reply:
changed